superdesk/media/media_editor.py

```diff
@@ -61,63 +60,97 @@ def transform(self, im, operation, param):
+            # first we get item and requested edit operations
             try:
-                item_id = request.json['item_id']
+                item = doc.pop('item')
             except KeyError:
-                raise errors.SuperdeskApiError.badRequestError('either item or item_id must be specified')
-            item = next(get_resource_service('archive').find({'_id': item_id}))
-        edit = request.json['edit']
-        rendition = item['renditions']['original']
-        media_id = rendition['media']
-        media = current_app.media.get(media_id)
-        out = im = Image.open(media)
-        for operation, param in edit.items():
-            try:
-                out = self.transform(out, operation, param)
-            except ValueError as e:
-                raise errors.SuperdeskApiError.badRequestError('invalid edit instructions: {msg}'.format(msg=e))
-        buf = BytesIO()
-        out.save(buf, format=im.format)
-        buf.seek(0)
-        content_type = rendition['mimetype']
-        ext = os.path.splitext(rendition['href'])[1]
-        filename = str(uuid.uuid4()) + ext
-        media_id = current_app.media.put(buf, filename=filename, content_type=content_type, folder='temp')
-        buf.seek(0)
-        renditions = generate_renditions(buf,
-                                         media_id,
-                                         [],
-                                         'image',
-                                         content_type,
-                                         get_renditions_spec(),
-                                         current_app.media.url_for_media,
-                                         temporary=True)
-        item['renditions'] = renditions
-        item[config.LAST_UPDATED] = utcnow()
-        docs[:] = [item]
-        return [item['_id']]
+                try:
+                    item_id = doc.pop('item_id')
+                except KeyError:
+                    raise errors.SuperdeskApiError.badRequestError('either item or item_id must be specified')
+                item = next(archive.find({'_id': item_id}))
+            else:
+                item_id = item[config.ID_FIELD]
+            edit = doc.pop('edit')
+
+            # new we retrieve and loag current original media
+            rendition = item['renditions']['original']
+            media_id = rendition['media']
+            media = current_app.media.get(media_id)
+            out = im = Image.open(media)
+
+            # we apply all requested operations on original media
+            for operation, param in edit.items():
+                try:
+                    out = self.transform(out, operation, param)
+                except ValueError as e:
+                    raise errors.SuperdeskApiError.badRequestError('invalid edit instructions: {msg}'.format(msg=e))
+            buf = BytesIO()
+            out.save(buf, format=im.format)
+
+            # we set metadata
+            buf.seek(0)
```

Comment:
May be it's better to hide in specific function ? like 
```
def set_metadata
```
And you make code more readeable than. And of course you don't need to write this comment

Reply:
Hi, thanks for your feedback.
Your suggestion make sense, but I'm not really fond of using methods for 3 lines, I find it more readable to see what's happening directly without having to jump in the code.
Question of taste I guess.